internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -129,6 +130,24 @@ func (store *updaterStore) AddDeploymentSpec(yamlDoc []byte) error {
 	if err := yaml.Unmarshal(yamlDoc, dep); err != nil {
 		return err
 	}
+
+	// OLM places the set of target namespaces for the operator in
+	// "metadata.annotations.olm.targetNamespaces". This value should be
+	// referred to in the Deployment spec's WATCH_NAMESPACE env.
+	for _, c := range dep.Spec.Template.Spec.Containers {
+		for _, e := range c.Env {
+			if e.Name == "WATCH_NAMESPACE" {
+				if e.ValueFrom == nil {
+					e.ValueFrom = &corev1.EnvVarSource{}
+				}
+				e.ValueFrom.FieldRef = &corev1.ObjectFieldSelector{
+					FieldPath: `metadata.annotations['olm.targetNamespaces']`,
+				}
+				break
+			}
+		}
+	}
```

Comment:
A couple of observations:

- If the user changes `deploy/operator.yaml` to specify WATCH_NAMESPACE with `value` instead of `valueFrom`, the final output seems to ignore our added `valueFrom` with the `metadata.annotations['olm.targetNamespaces']` field path. It seems like whatever serializes the pod spec favors `value` over `valueFrom`.

- Also, if the user has modified their `deploy/operator.yaml` and removed `WATCH_NAMESPACE`, we'll do nothing.

It seems like we should probably handle both of these edge cases.

- For the first case, should we just flat out error out with a message that says OLM doesn't support hardcoding the `WATCH_NAMESPACE`?

- For the second case, it could be that the operator developer wants to use another environment variable name for `WATCH_NAMESPACE`. Should we log a warning with something along the lines of:

    - `No WATCH_NAMESPACE environment variable detected in operator deployment. For OLM compatibility, your operator MUST watch namespaces defined in metadata.annotations['olm.targetNamespaces']`

Reply:
Conclusion of offline discussion (from @joelanford):

1. If there’s a `WATCH_NAMESPACE` env always overwrite it in the CSV to use `olm.targetNamespaces`.
2. If there’s no `WATCH_NAMESPACE` env, check if `olm.targetNamespaces` is references in the deployment podSpec in places where the downward API is supported. If present, copy everything over verbatim. If not present, log a warning.